src/Compiler.php

```diff
@@ -114,18 +121,28 @@ private function getArrayProperty()
         $token = array_shift($this->request);
 
         if (!self::scanProperty($token, $array)) {
-            return false;
+            throw new Exception(
+                self::ERROR_NO_INITIAL_PROPERTY,
+                array('token' => $token),
+                self::ERROR_NO_INITIAL_PROPERTY .
+                " Error: API requests must begin with a property."
+            );
         }
 
         list($class, $path) = $this->schema->getPropertyDefinition('Database', $array);
-        $list = array_pop($path);
 
-        if (!isset($class, $path, $list)) {
+        if (!isset($class, $path)) {
             return false;
```

Comment:
Now that we're using exceptions, this function should always throw an exception (rather than silently return a false value, as it used to in the past).